controllers/grid/issues/form/IssueForm.php

```diff
@@ -241,9 +241,16 @@ public function execute(...$functionArgs)
         $issue->setVolume(empty($volume) ? null : $volume);
         $issue->setNumber(empty($number) ? null : $number);
         $issue->setYear(empty($year) ? null : $year);
-        if (!$isNewIssue) {
+
+        // If issue is not published, allow empty datePublished
+        if (!$this->getData('datePublished') && !$issue->getPublished()) {
+            $issue->setDatePublished(null);
+        }
```

Comment:
Do you need to add a check to the `validate()` method when `!$this->getData('datePublished') && $issue->getPublished()`?

Reply:
You mean if no `datePublished` is given, but `issue` is published, we should give a warning. I think at the moment you can't just remove the date and no warning is shown.

Maybe we need a check for a valid date as well? We did not have one formerly, but would probably make sense. I was just looking around and could not find if we have one somewhere already, I mean if there is an existing solution for validating date stamps.

Reply:
Yeah, so if someone removes the `datePublished` and clicks save, and the `datePublished` isn't removed, they need to see some message indicating that. Otherwise they will assume it's changed and then we have bad metadata.

I don't think we have one of the old form validators, but if you look at `ValidatorRegExp` you can see an example of using [Laravel's Validation Rules](https://laravel.com/docs/9.x/validation#available-validation-rules). I think we'd want [date_format:format](https://laravel.com/docs/9.x/validation#rule-date-format), something like this:

 ```php
$validator = ValidatorFactory::make(
    ['value' => $value],
    ['value' => ['required', 'date_format:Y-m-d']]
);

$result = $validator->passes();
```

You can also skip the `Validator` class and just add this to `IssueForm::validate()`.

Reply:
The `datePublished` field is acting very weird. I think the `datepicker` functionality is affecting it.

I added `error_log(print_r($this->getData('datePublished'), true));` to the form `validation` method to see what I get there when I save the form.

```
Form input: (empty) -> Error output (empty) (this value is saved to the issue!)
Form input: 55 -> Error output (empty) (issue value still empty)
Form input: 2019-01 -> Error output (empty)  (issue value still empty)
Form input: 2019-01-01 -> Error output 2019-01-01 (this value is saved to the issue!)
Form input: 55 -> Error output 2019-01-01   (issue value still  2019-01-01)
Form input: 2019-01 -> Error output 2019-01-01 (issue value still  2019-01-01)
Form input: (empty) -> Error output (empty) (this value is saved to the issue!)
```

So it looks like there is some internal validation going on in the datepicker?

Reply:
There may be client-side validation, but it can't replace server-side validation.

Reply:
yes, looking at the source with developer tools I can see that the `value` attribute in `input` is only updated when you enter a valid date.

Reply:
so basically you can not get an invalid date from the form when you get to validation. The problem is of course that you won't see any error. It will just use the last valid value it had.

Reply:
You can't rely on the UI to do validation because the user can send any data in the HTTP request.

Reply:
sure, but you can not get an invalid date at the moment to the validation method.

Reply:
I had something like

```
        if ($this->getData('datePublished')) {

            $validator = ValidatorFactory::make(
               ['value' => $this->getData('datePublished')],
               ['value' => ['required', 'date_format:Y-m-d']]
            );

            if (!$validator->passes()){
                $this->addError('datePublished', __('editor.issues.datePublished.invalid'));
                $this->addErrorField('datePublished');
            }
        }
```

in the validation, but the error is never triggered because the date is always valid.